Test/ElasticScale.ShardManagement.UnitTests/ShardMapperTests.cs

```diff
@@ -1544,6 +1644,105 @@ public void KillConnectionOnOfflineRangeMapping()
             }
         }
 
+        /// <summary>
+        /// Take a mapping offline, verify that the existing connection is not killed using mapping options.
+        /// </summary>
+        [TestMethod()]
+        [TestCategory("ExcludeFromGatedCheckin")]
+        public void DoNotKillConnectionOnOfflineRangeMapping()
```

Comment:
These 2 tests are very similar. Can you pull out the common code into a method? I understand that it's not easy since ListShardMap and RangeShardMap don't have a shared interface, but it should be possible. Some ideas that might work:
1. the test could pass an `Action` callback that marks offline
2. just bypass type checker by using `dynamic`